proxy/src/bind.rs

```diff
@@ -322,19 +321,4 @@ impl Protocol {
 
         Protocol::Http1(host)
     }
-
-    pub fn is_cachable(&self) -> bool {
-        match *self {
-            Protocol::Http2 | Protocol::Http1(Host::Authority(_)) => true,
-            _ => false,
-        }
-    }
-
-    pub fn into_key<T>(self, key: T) -> Reuse<(T, Protocol)> {
```

Comment:
I think removing this logic from `Bind` is a good call. At the time, it felt like a good reduction in code duplication, but looking back, it definitely introduces some unnecessary complexity. 👍